CartoDB/cartodb
# Fix exception handling

Fixes https://rollbar.com/carto/CartoDB/items/41419/

Properly handles error when calling `optional_api_authorization` with old session.

**How to test this?**

In staging this should return a proper 401:

```
curl 'https://ayuntamiento-banastas.carto-staging.com/u/amiedes/api/v3/me' \
  -H 'Connection: keep-alive' \
  -H 'Pragma: no-cache' \
  -H 'Cache-Control: no-cache' \
  -H 'Accept: */*' \
  -H 'X-Requested-With: XMLHttpRequest' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36' \
  -H 'Content-Type: application/json; charset=utf-8' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Referer: https://ayuntamiento-banastas.carto-staging.com/u/amiedes/profile' \
  -H 'Accept-Language: es-ES,es;q=0.9,en;q=0.8' \
  -H 'Cookie: _ga=GA1.2.430590267.1588757521; _gid=GA1.2.1132459692.1588757521; _fbp=fb.1.1588839579399.1826356413; _gat_UA-20934186-25=1; _gat_UA-20934186-31=1; _cartodb_base_url=https%3A%2F%2Fayuntamiento-banastas.carto-staging.com%2Fu%2Famiedes; _cartodb_session=OEFJM0lXVWwrNVNZanlIYUNERlJXZW9RN2YyamtPdnFzSENScmNXcHpJemRYYUVKZDBBNWRESjhxdUlzRmtLa3RpYjhPaVZmMk9Vd1ZJR0ozdGRZNXJPYkExWEY5MXBvUjA2S3grYzlBYW4zQzAvZWNpZk9BdTBNMWoyRWJXQ2swaTFPZXVpOGFiWUI1dmgveFoxTlBXSXEvQzAwQ2oxUG9vM3h6OFF0UDc4YWc0VWs2QUZTQk5aL0h0VHpSbytMUGNIQzhpWkdQL1pNbmNjeEJ4NnJ1RSs5azJMSTFPT2tFTkZSUzYzbHZudjdrbGdMZ3dpSFR2WGVjZ2FkdTgzTGhTTTRzaXZ6OFBIV3BkQmtLbHNocWxlSXRPb0x3S1p3WTI0a2p4WVp5RmtBbGs5cWErTy9MN0MvNWgraU51ZHNQbHkxTW9CUmlWT1hSY1EwSThDTTRzNFlTV2FQd3lCcVh2S09sL0Y4MzJWZW5zZW5zU2k2VzJtNGR1R3N2a0FQNW45R3dXOEVRem45Zzg0RU5URENha3ZxQUVDN1pUUVNTMnI3SWQwOW9rUkJTcFR6RlhhVmxrZWsycklzSDRFMTBhY3hwRFJrRzhkNmNFVFJOV0FKUk9kRXhPRzA4NmFvNGxNbFUvUklIZ3pjZ01PSFpDR0tGNFUvME5hZkpxL3JBSERLWS9uejBCdUhoV21NdnBOenk4OFRaYkNsRWJzT1l6aFdXQlUvd2x1ZlhFRDRjcWliVmljRjNhd0txSjdjYXJRWE1FcjhudmdLNk9Oekhyc3hwQT09LS1ZNG5FanJWOTNSVlRIbEVnRlVQd2tBPT0%3D--210f4815813fdc4eb9a5bca87db3fd00ba9b8ce4' \
  --compressed -v
```
